addons/mrp/models/mrp_bom.py

```diff
@@ -288,7 +288,7 @@ def _get_default_product_uom_id(self):
         return self.env['uom.uom'].search([], limit=1, order='id').id
 
     product_id = fields.Many2one('product.product', 'Component', required=True, check_company=True)
-    product_tmpl_id = fields.Many2one('product.template', 'Product Template', related='product_id.product_tmpl_id')
+    product_tmpl_id = fields.Many2one('product.template', 'Product Template', related='product_id.product_tmpl_id', store=True, index=True)
```

Comment:
I would have choose the other solution for master only (don't change the compute and add index on product_id) for multiple reason: 
- We are more likely to used the inverse of `product_id` in product (`bom_line_ids`) (actually it is already the case in mps).
- Avoid duplicate the same information on table which can be big.

The performance shouldn't be impact a lot (To test) even with product with a lot of variant.

Reply:
When I tried to add index to product_id in my case, it was twice speed up only (350 ms vs 700 ms)

Reply:
Not very efficient. Then it is ok for me.

Reply:
For the product.product model, the `used_in_bom_count` field, it can be a bottleneck too, no ?

Reply:
> For the product.product model, the `used_in_bom_count` field, it can be a bottleneck too, no ?

I didn't measure, but it should be fine: in product.product you don't need to make sub select to fetch 1000 variant ids and filter mrp lines by them

Reply:
Yes right, I forgot this point.

Reply:
r+ ?